sys/net/include/netdev.h

```diff
@@ -0,0 +1,262 @@
+     *
+     * @return  0, on success
+     * @return  -ENODEV, if *dev* is not recognized
+     */
+    int (*rem_receive_data_callback)(netdev_t *dev, netdev_rcv_data_cb_t cb);
+
+    /**
+     * @brief   Get an option value from a given network device.
+     *
+     * @param[in] dev           the network device
+     * @param[in] opt           the option type
+     * @param[out] value        pointer to store the gotten value in
+     * @param[in,out] value_len the length of *value*. Must be initialized to the
+     *                          available space in *value* on call.
+     * @return  0, on success
+     * @return  -ENODEV, if *dev* is not recognized
+     * @return  -ENOTSUP, if *opt* is not supported for the device with this
+     *          operation
+     * @return  -EOVERFLOW, if available space in *value* given in *value_len*
+     *          is not big enough to store the option value.
+     * @return  any other fitting negative errno if the ones stated above
+     *          are not sufficient
+     */
+    int (*get_option)(netdev_t *dev, netdev_opt_t opt, void *value,
+                      size_t *value_len);
+
+    /**
+     * @brief   Set an option value for a given network device.
+     *
+     * @param[in] dev           the network device
+     * @param[in] opt           the option type
+     * @param[in] value         the value to set
+     * @param[in] value_len     the length of *value*
+     *
+     * @return  0, on success
+     * @return  -EINVAL, if *value* is not in a required format
+     * @return  -ENODEV, if *dev* is not recognized
+     * @return  -ENOTSUP, if *opt* is not supported for the device with this
+     *          operation
+     * @return  -EOVERFLOW, if *value_len* is larger then the device expects.
+     * @return  -EPROTONOSUPPORT, if *opt* was NETDEV_OPT_PROTO and type is
+     *          not supported.
+     * @return  any other fitting negative errno if the ones stated above
+     *          are not sufficient
+     */
+    int (*set_option)(netdev_t *dev, netdev_opt_t opt, void *value,
+                      size_t value_len);
+
+    /**
+     * @brief   Get state from a given network device.
+     *
+     * @param[in] dev       the network device
+     * @param[out] state    the network device
+     *
+     * @return  0, on success
+     * @return  -ENODEV, if *dev* is not recognized
+     * @return  -ETIME, if device timed out on trying to acquire state
+     */
+    int (*get_state)(netdev_t *dev, netdev_state_t *state);
+
+    /**
+     * @brief   Set state from a given network device.
+     *
+     * @param[in] dev      the network device
+     * @param[in] action   action to trigger
+     *
+     * @return  0, on success
+     * @return  -ENODEV, if *dev* is not recognized
+     * @return  -ENOTSUP, if *state* is not supported
+     * @return  -ETIME, if device timed out on trying to change state
+     */
+    int (*trigger)(netdev_t *dev, netdev_action_t action);
```

Comment:
When going through the use case for pre-loading, I have the impression, that "triggering an action" is maybe more straight forward than "setting a state". So triggering the command: "send now" is more intuitive than putting the device into the state "TX". As for the command one expects the device to return to whatever state it was in before, while setting the state to "TX" we would expect the device to stay there. So I think triggering actions is more versatile and understandable then setting a state.